R/utils.R

```diff
@@ -167,34 +167,70 @@ repair_variable_names <- function(names) {
   names
 }
 
-
-
-
 #' Dump data to temporary file in format readable by CmdStan
 #'
-#' Currently calls `rstan::stan_rdump()` to create the `.data.R` file.
-#' FIXME:
-#'
 #' @param data A named list of \R objects.
 #' @return Path to temporary file containing the data.
 #' @noRd
-write_rdump <- function(data) {
-  # FIXME don't use rstan
-  if (!requireNamespace("rstan", quietly = TRUE)) {
-    stop("Please install the 'rstan' package. This is temporarily required ",
-         "for writing the data file for CmdStan.", call. = FALSE)
+write_json <- function(data) {
+  if (!requireNamespace("jsonlite", quietly = TRUE)) {
+    stop("Please install the 'jsonlite' package.", call. = FALSE)
   }
   checkmate::assert_names(names(data), type = "unique")
-  temp_file <- tempfile(pattern = "standata-", fileext = ".data.R")
-  rstan::stan_rdump(
+  temp_file <- tempfile(pattern = "standata-", fileext = ".json")
+  cmdstanr_jsondump(
     list = names(data),
     file = temp_file,
     envir = as.environment(data)
   )
   temp_file
 }
 
-
+#' Dump data to temporary file in format readable by CmdStan
+#'
+#' @param list A vector of character string: the names of one or more R objects to be dumped.
+#' @param file A character string represeneting the path to the output file
+#' @param envir The environment to search for the listed objects.
+#' @export
+#' @return Path to temporary file containing the data.
+cmdstanr_jsondump <- function(list,
+                          file,
+                          envir = parent.frame()) {
+  if (!requireNamespace("jsonlite", quietly = TRUE)) {
+    stop("Please install the 'jsonlite' package.", call. = FALSE)
+  }
+  temp_file <- tempfile(pattern = "standata-", fileext = ".json")
+  # check filename is valid(vector of characters) and of nonzero length
+  if (!is.character(file) || !nzchar(file)) {
+    warning("The supplied filename is invalid!")
+    return(invisible(character()))
+  }
```

Comment:
I know stan_rdump does warnings and returns empty character() objects, but unless you like that, I'd prefer just throwing an error here. That's probably what stan_rdump should be doing too but it's a really old function and hasn't been refactored in ages.

Reply:
Tried to resemble the behaviour of rstan again here. Will change back. Agree with throwing error here.